thesis/03-implementation.tex

```diff
@@ -171,6 +171,352 @@ \subsection{Components overview}
+
+where $l$ is the current edge length and $k_{axial}$ is a constant.
+
+\smallskip
+Before we apply the force to the vertices, first we need to convert it to a vector $\pmb{F}_{beam}$ in a model's coordinate space.
+The $\pmb{F}_{beam}$ is related to the vertices position $\pmb{p}$ by:
+
+$$\pmb{F}_{beam} = -\nabla V(\pmb{p}) = -\frac{\partial V}{\partial \pmb{p}}$$
+
+where V is the potential energy of the system. By the chain rule, we get:
+
+\begin{equation} \label{solver:beam_force}
+\pmb{F}_{beam} = -\frac{\partial V}{\partial l}\frac{\partial l}{\partial \pmb{p}} = F_{l}\frac{\partial l}{\partial \pmb{p}} = -k_{axial} * (l - l_{0})\frac{\partial l}{\partial \pmb{p}}
+\end{equation}
+
+
+For the two vertices attached to the edge, we have:
+
+$$ \frac{\partial l}{\partial \pmb{p_{1}}} = -\uveci_{12}, \frac{\partial l}{\partial \pmb{p_{2}}} = \uveci_{12} $$
+
+where
+\begin{itemize}
+	\item $\uveci_{12}$ is a unit vector from vertex 1 to vertex 2
+	\item $\pmb{p_{1}}$ is vertex 1 position
+	\item $\pmb{p_{2}}$ is vertex 2 position
+\end{itemize}
+
+Although axial stiffness $k_{axial}$ could be related to material properties, we devised it experimentally.
+
+All of this boils down to a few lines of code:
+
+\lstinputlisting[language=Python]{assets/listings/beam_force.py}
+
+
+\medskip
+\textbf{Crease constraints - crease force}
+
+With axial constraints we restrict the model to retain its shape while folding.
+However, the main force that drives folding process is the result of crease constraints.
+\smallskip
+
+We drive faces connected by an edge to fold towards some target fold angle $\theta$.
+The angle $\theta$ is the supplement of the dihedral angle between the two neighbouring triangular faces.
+
+\begin{figure}[H]
+	\caption{Crease constraints formulation between the two neighbouring triangular faces}
+    \centering
+	\includegraphics[width=.6\linewidth]{assets/3-crease_force_face.png}
+	\label{solver:crease_force_face}
+\end{figure}
+
+Crease constraints are modeled as linear-elastic torsional springs.
+Therefore, similarly to equation \eqref{solver:beam_force}, crease constraints will apply forces to the neighbouring
+vertices by:
+
+\begin{equation} \label{solver:crease_force}
+	\pmb{F}_{crease} = -k_{crease}(\theta - \theta _{target})\frac{\partial \theta}{\partial \pmb{p}}
+\end{equation}
+
+where:
+
+\begin{itemize}
+	\item $\pmb{F}_{crease}$ is a force applied to a vertex in a model's coordinate space
+	\item $k_{crease}$ is an edge folding stiffness (more on that below)
+	\item $\theta$ is the current angle between the two faces (strictly speaking, supplement of the dihedral angle between the faces)
+	\item $\theta _{target}$ is the desired angle between the two faces
+\end{itemize}
+
+In the implementation, there are 5 possible edge assignments:
+
+\begin{enumerate}
+	\item valley
+	\item mountain
+	\item flat
+	\item boundary
+	\item unknown
+\end{enumerate}
+
+The $k_{crease}$ depends on the type of the edge.
```

Comment:
`the type of the edge`, I think the second `the` is redundant/invalid here. The reference is not to a particular edge, but an edge in general.

Not sure if the first `the` should also not be changed to `an`.